mlrun/runtimes/nuclio/serving.py

```
@@ -812,3 +821,31 @@ def _set_as_mock(self, enable):
             "Turn off the mock (mock=False) and make sure Nuclio is installed for real deployment to Nuclio"
         )
         self._mock_server = self.to_mock_server()
+
+    def to_job(
+        self, target_mapping: Optional[dict[str, TaskStep]] = None
+    ) -> KubejobRuntime:
+        """
+        Convert this ServingRuntime to a KubejobRuntime, so that the graph can be run as a standalone job.
+
+        :param target_mapping: Set this to map targets in this ServingRuntime to other targets that will run as
+          part of the standalone job. For example, you may wish to replace a ParquetTarget in order to change its path.
+        """
+        serving_job = self.copy()
+        if target_mapping:
+            for from_name, (to_name, to_step) in target_mapping.items():
```

So you're assuming that `target_mapping` is a dict `str -> tuple(name, step)`? This doesn't match the definition of the object in the function signature. Also, not sure why you need the `to_name` in this case? The thinking was that the step provided will have the same target name as the one it replaces.

---

Removed `to_name`.

> The thinking was that the step provided will have the same target name as the one it replaces.

Regarding this one, should we enforce that? I.e. fail if the step's name (as defined in step.name) differs from the original name?

This could prevent the pitfall of breaking steps that come after the target, which storey supports in principle, but is not a typical use case and is not even very well-defined in terms of what events a target step might output and with what guarantees.

---

I think we should enforce. Failing early at this point will save trouble later. So, verify that the step sent as target has the same name as the one being replaced.

---

Done.